youpin-api.js

```diff
@@ -20,21 +17,17 @@ class Api {
       .configure(authentication());
 
     setInterval(() => {
-      api.refreshToken();
+      Api.refreshToken();
     }, 23 * 60 * 60000); // Refresh token every 23 hours
 
-    return new Promise((resolve, reject) => {
-      app.authenticate({
-        type: 'local',
-        'email': username,
-        'password': password
-      }).then(result => {
-        this.token = app.get('token');
-        resolve(this);
-      }).catch(error => {
-        console.log(error);
-        reject(error);
-      });
+    app.authenticate({
+      type: 'local',
+      email: this.username,
+      password: this.password,
+    }).then(() => {
+      this.token = app.get('token');
```

Comment:
I already checked it. The constructor needs to `return new Promise` because it will be use at [app.js](https://github.com/iswanj/youpin-bot/blob/6e76a4e6bf7e95eff1c391e0da243b430fe72104/app.js#L58).

In this case, I think it would pass eslinting. So, @iswanj could you wrap with new Promise back and restore `resolve(this)` and `reject(error)`.